homeassistant/components/unmanic/__init__.py

```diff
@@ -0,0 +1,115 @@
+"""The Unmanic integration."""
+import asyncio
+from datetime import timedelta
+import logging
+from typing import List
+
+from unmanic_api import Unmanic
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_TIMEOUT,
+    CONF_VERIFY_SSL,
+    __version__,
+)
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, PLATFORMS
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup(hass: HomeAssistant, config: Config):
+    """Set up this integration using YAML is not supported."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up this integration using UI."""
+    if hass.data.get(DOMAIN) is None:
+        hass.data.setdefault(DOMAIN, {})
+
+    session = async_get_clientsession(hass)
+    client = Unmanic(
+        host=entry.data[CONF_HOST],
+        port=entry.data[CONF_PORT],
+        request_timeout=entry.data[CONF_TIMEOUT],
+        session=session,
+        tls=entry.data[CONF_SSL],
+        verify_ssl=entry.data[CONF_VERIFY_SSL],
+        user_agent=f"HomeAssistant/Unmanic/{__version__}",
+    )
+
+    coordinator = UnmanicUpdateCoordinator(hass, client=client)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    for platform in PLATFORMS:
+        if entry.options.get(platform, True):
+            coordinator.platforms.append(platform)
+            hass.async_add_job(
+                hass.config_entries.async_forward_entry_setup(entry, platform)
+            )
+
+    entry.add_update_listener(async_reload_entry)
+    return True
+
+
+class UnmanicUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from the API."""
+
+    def __init__(self, hass: HomeAssistant, client: Unmanic) -> None:
+        """Initialize."""
+        self.api = client
+        self.platforms = []  # type: List[str]
+
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+
+    async def _async_update_data(self):
+        """Update data via library."""
+        try:
+            data = {}
+            data["pending_tasks"] = await self.api.get_pending_tasks()
+            data["settings"] = await self.api.get_settings()
+            data["task_history"] = await self.api.get_task_history()
+            data["version"] = await self.api.get_version()
+            data["workers_status"] = await self.api.get_workers_status()
+            return data
+        except Exception as exception:
```

Comment:
Do you have any other (more specific) exceptions that you can catch? I don't think you need to catch a broad exception in your `_async_update_data`, since this the Data Update Coordinator takes care of this already.